frontend/src/components/material/MaterialTable.vue

```diff
@@ -363,6 +363,15 @@ export default {
         .catch(error => {
           this.$set(this.newMaterialItems[key], 'serverError', error)
         })
+    },
+
+    async findOneActivityByContentNode (contentNode) {
+      await this.camp.activities().$loadItems()
+      const root = await contentNode.$href('root')
+
+      return this.camp.activities().items.find(activity => {
+        return activity.rootContentNode()._meta.self === root
```

Comment:
Not in DB, no.

The protection is "only" from API side, because `rootCotentNode` is not writable and always a new one is created, when an activity or a category is created.

Reply:
So, could we add a unique index in the DB to the `activity#root_content_node_id` and `category#root_content_node_id` DB columns? Do you see any drawbacks with that?

Reply:
Is Doctrine and Postgres able to handle unique constraints across multiple tables? If so, sure. No drawback I see.

Reply:
Discussion: Add at least a unique index to each table (activity + category)

Reply:
This was already implemented. See `unique: true` on `JoinColumn` Attribute:
https://github.com/ecamp/ecamp3/pull/2683/files#diff-6c5bf17f3740f1a2190f671331eb17d4f1b4ea14ff04d2883f63f3ec379093a1L20

and the actual indexes in the migration files:
https://github.com/ecamp/ecamp3/pull/2683/files#diff-7f44091fbb4175265ebac8238f7aebed816b76abaf3bbcf6a153a0eb46413cecR38

https://github.com/ecamp/ecamp3/pull/2683/files#diff-7f44091fbb4175265ebac8238f7aebed816b76abaf3bbcf6a153a0eb46413cecR46